### ISSUE TITLE
`--select` and `--ignore` have no effect on rule filtering

### ISSUE THREAD

**Issue by campbell-brown** [swanysimon/mdlint#68]
# Description

Neither `--select <RULE_CODE>` nor `--ignore <RULE_CODE>` change which rules are reported by `mdlint check`. The full default rule set runs regardless of what's passed to either flag.

# Steps to reproduce

`mdlint` version: 0.3.22

Given this markdown file:

```markdown
# Example

This is a deliberately long line that exceeds the default one hundred and twenty character line length limit for sure, no really, it does.
```

Run:

```shell
mdlint check --select MD001 example.md
mdlint check --ignore MD013 example.md
```

# 	Actual

`--select MD001` detects the `MD013` rule, even though I've explicitly set `MD001`:

```
example.md
  example.md:3:121: MD013 Line exceeds maximum length (138 > 120)
       | This is a deliberately long line that exceeds the default one hundred and twenty character line length limit for sure, no really, it does.
       |                                                                                                                         ^

Found 1 error(s) in 1 file(s) (1 checked)
```

`--ignore MD013` detects the `MD013` rule, even though I've explicitly ignored it:

```
example.md
  example.md:3:121: MD013 Line exceeds maximum length (138 > 120)
       | This is a deliberately long line that exceeds the default one hundred and twenty character line length limit for sure, no really, it does.
       |                                                                                                                         ^

Found 1 error(s) in 1 file(s) (1 checked)
```

## Expected

`--select MD001` only reports `MD001` violations, `--ignore MD013` reports everything except `MD013` violations.

# Workaround

Using a `mdlint.toml` works properly:

```toml
[rules.MD013]
enabled = false
```

```
❯ mdlint check example.md
Checked 1 file(s), no errors found.
```
and:
```toml
default_enabled = false

[rules.MD001]
enabled = true
```
```
❯ mdlint check example.md
Checked 1 file(s), no errors found.
```

**Comment by swanysimon** [comment]
Oh, that's a bad regression, damn. Lookin'

**Comment by swanysimon** [comment]
@campbell-brown thanks for all the issues, I think Claude and I got through them all, and I'll consolidate to this thread. Check 0.3.23 and let me know if you find any other issues
